public/app/features/templating/template_srv.ts

```diff
@@ -258,46 +264,85 @@ export class TemplateSrv {
     return scopedVar.value;
   }
 
-  replace(target: string, scopedVars?: ScopedVars, format?: string | Function): any {
+  replace(target: string, scopedVars: ScopedVars, format: 'list'): string[];
+  replace(target: string, scopedVars?: ScopedVars, format?: string | Function): string;
+  replace(target: string, scopedVars?: ScopedVars, format?: string | Function): string | string[] {
```

Comment:
what is this syntax?  (I have not seen it before... what does it do?)  Is it the same as:
```
replace(target: string, scopedVars?: ScopedVars, format?: 'list' | string | Function): string | string[] {
```

Reply:
It is using [TypeScript Function Overloading](https://www.typescriptlang.org/docs/handbook/functions.html#overloads). This is more precise than 

     replace(target: string, scopedVars?: ScopedVars, format?: 'list' | string | Function): string | string[] {

since it specifically ensures it returns a string array only if format is "list".

Reply:
this feels like it should be another function and not just a format, format just change interpolation

Reply:
@torkelo What should be the name of the new function? Just `list` as in

`list(target: string, scopedVars: ScopedVars): string[];`

or  maybe `allValues`?